tests/ui/macros/rfc-3086-metavar-expr/syntax-errors.stderr

```diff
@@ -313,6 +373,23 @@ LL |     unknown_metavar!(a);
    |
    = note: this error originates in the macro `unknown_metavar` (in Nightly builds, run with -Z macro-backtrace for more info)
 
+error: `${concat(..)}` currently only accepts identifiers as parameters
+  --> $DIR/syntax-errors.rs:162:19
+   |
+LL |         ${concat($ex, aaaa)}
+   |                   ^^
+
+error: expected identifier, found `$`
+  --> $DIR/syntax-errors.rs:178:15
+   |
+LL |         const ${concat($sign name, _123)}: () = ();
+   |               ^ expected identifier
+...
+LL |     tt_that_is_dollar_sign_with_concat!($, FOO);
+   |     ------------------------------------------- in this macro invocation
+   |
+   = note: this error originates in the macro `tt_that_is_dollar_sign_with_concat` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment:
This diagnostic isn't originated from `parse_ident`. Mabe something after expansion?

If a better span is desired, then a separated investigation should be created.

Reply:
IFAICT, this specific diagnostic wasn't emitted by `parse_ident`.

But regardless `parse_ident` was modified to use as much span from tokens as possible. The outside span is still being used in case the `RefTokenTreeCursor` iterator doesn't yield.